provider/terraform/config.rb

```diff
@@ -29,6 +29,14 @@ def resource_override
       def property_override
         Provider::Terraform::PropertyOverride
       end
+
+      def new_resource_override
```

Comment:
This is called `new_` because these are the new overrides right? I would comment explaining that because the function name alone suggests that this will make a new resource override.

Also are there a lot of calls to `resource_override` or can we start using that name for the new ones now and call the old ones `old_resource_override`?

Reply:
I didn't want to touch the old overrides because this is already a long PR. The `new` part does refer to the new overrides. I'll add a comment.

Reply:
sgtm